packages/dd-trace/src/exporters/common/agents.js

```diff
@@ -0,0 +1,42 @@
+'use strict'
+
+const http = require('http')
+const https = require('https')
+const { storage } = require('../../../../datadog-core')
+
+const keepAlive = true
+const maxSockets = 1
+
+function createAgentClass (BaseAgent) {
+  class CustomAgent extends BaseAgent {
+    constructor () {
+      super({ keepAlive, maxSockets })
```

Comment:
Should be passing the given options rather than using your own like this. If anything tries to create a pool with multiple sockets it could break when it tries to acquire those multiple sockets and is only able to get one.

Reply:
This should not be required as the goal of the `request` module is explicitly to manage that. Also, this is not a change from before since it was already hardcoded, so it's preserving the existing behaviour. If we want to change this behaviour it should be a separate PR with proper justification.

Reply:
Ah, sorry, misread this as an instrumentation for http.Agent. Ignore me! 😅